crates/core/tedge_mapper/src/c8y/actor.rs

```diff
@@ -0,0 +1,314 @@
+            let _ = self.mqtt_publisher.send(converted_message).await;
+        }
+
+        Ok(())
+    }
+
+    async fn process_file_watch_event(
+        &mut self,
+        file_event: FsWatchEvent,
+    ) -> Result<(), RuntimeError> {
+        match file_event.clone() {
+            FsWatchEvent::DirectoryCreated(path) => {
+                if let Some(directory_name) = path.file_name() {
+                    let child_id = directory_name.to_string_lossy().to_string();
+                    let message = Message::new(
+                        &Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC),
+                        format!("101,{child_id},{child_id},thin-edge.io-child"),
+                    );
+                    self.mqtt_publisher.send(message).await?;
+                }
+            }
+            FsWatchEvent::FileCreated(path)
+            | FsWatchEvent::FileDeleted(path)
+            | FsWatchEvent::Modified(path)
+            | FsWatchEvent::DirectoryDeleted(path) => {
+                match process_inotify_events(&path, file_event) {
+                    Ok(Some(discovered_ops)) => {
+                        self.mqtt_publisher
+                            .send(
+                                self.converter
+                                    .process_operation_update_message(discovered_ops),
+                            )
+                            .await?;
+                    }
+                    Ok(None) => {}
+                    Err(e) => {
+                        eprintln!("Processing inotify event failed due to {}", e);
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub async fn process_sync_timeout(&mut self) -> Result<(), RuntimeError> {
+        // Once the sync phase is complete, retrieve all sync messages from the converter and process them
+        let sync_messages = self.converter.sync_messages();
+        for message in sync_messages {
+            self.process_mqtt_message(message).await?;
+        }
+
+        Ok(())
+    }
+}
+
+pub struct C8yMapperBuilder {
+    config: C8yMapperConfig,
+    box_builder: SimpleMessageBoxBuilder<C8yMapperInput, C8yMapperOutput>,
+    mqtt_publisher: Option<DynSender<MqttMessage>>,
+    http_proxy: Option<C8YHttpProxy>,
+    timer_sender: Option<DynSender<SyncStart>>,
+}
+
+impl C8yMapperBuilder {
+    //HIPPO: Accept all the providers as arguments
+    pub fn new(config: C8yMapperConfig) -> Self {
+        let box_builder = SimpleMessageBoxBuilder::new("CumulocityMapper", 16);
+
+        Self {
+            config,
+            box_builder,
+            mqtt_publisher: None,
+            http_proxy: None,
+            timer_sender: None,
+        }
+    }
+
+    pub fn with_c8y_http_proxy(
+        &mut self,
+        http: &mut impl ServiceProvider<C8YRestRequest, C8YRestResult, NoConfig>,
+    ) -> Result<(), LinkError> {
+        self.http_proxy = Some(C8YHttpProxy::new("C8yMapper => C8Y", http));
```

Comment:
Is this the right name `"C8yMapper => C8Y",`

Reply:
It could be renamed to "C8yMapper => C8yHttpProxy"